core/plannernotes.c

```diff
@@ -449,14 +449,16 @@ void add_plan_to_notes(struct diveplan *diveplan, struct dive *dive, bool show_d
 	for (int gasidx = 0; gasidx < MAX_CYLINDERS; gasidx++) {
 		double volume, pressure, deco_volume, deco_pressure, mingas_volume, mingas_pressure, mingas_d_pressure, mingas_depth;
 		const char *unit, *pressure_unit, *depth_unit;
-		char warning[1000] = "";
-		char mingas[1000] = "";
+		char *warning;
+		char *mingas;
 		cylinder_t *cyl = &dive->cylinder[gasidx];
 		if (cyl->cylinder_use == NOT_USED)
 			continue;
 		if (cylinder_none(cyl))
 			break;
 
+		asprintf(&warning, "");
```

Comment:
This allocates an empty string that is not freed in the case of other asprintf calls.
If you really want to write it that way, you could do a `warning = strdup("");` and a `free(warning)` before additional `asprintf` calls.

But really, I'd prefer finding the source of the original bug. Is 1000 too small a buffer size?